src/library_pthread.js

```diff
@@ -320,17 +328,16 @@ var LibraryPThread = {
             } else if (d.cmd === 'alert') {
               alert('Thread ' + d.threadId + ': ' + d.text);
             } else if (d.cmd === 'exit') {
-              // Thread is exiting, no-op here
+              var detached = Atomics.load(HEAPU32, (d.threadId + {{{ C_STRUCTS.pthread.detached }}} ) >> 2);
```

Comment:
Instead of passing a `d.threadId`, the thread ID of the thread that the given current can be located with `worker.pthread.thread` - that is nicer since it retains more coherency, and less state to postMessage over. I now see that the `alert` command does also pass a `threadId` message (that is actually the only command that does past the `threadId` as custom, but it should not need to, for the same reason)

Reply:
Ah good point. I did this before writing the `returnThreadToPool` function, so I didn't realise that at the time. I just followed how the above messages behave. `print`, `printErr` and `alert` all take a `d.threadId`. I'm not going to change that in this PR, but might open another one to address this.